machine.go

```diff
@@ -133,9 +133,9 @@ func (cfg *Config) Validate() error {
 type Machine struct {
 	// Handlers holds the set of handlers that are run for validation and start
 	Handlers Handlers
+	Client   *Client
```

Comment:
Why is client made public now?

Reply:
Client needs to be public since there isn't a way to call things like `PutMmds` due to removal of `SetMetadata`. It should be public if users want to use the higher level client we've created without the need of creating one when they already have a `Machine` object.

Reply:
The whole point of the the SDK is to isolate the the generated interface behind an abstraction layer. I don't think we want to be exposing these details, and I certainly don't think we want to force the user to interact directly with the generated code in order to access the complete functionality.

Also, this is a breaking change, so if we're going to remove SetMetadata, we should be careful about it.

Reply:
Yea, this is a breaking change, but having two methods, `EnableMetadata` and `SetMetadata` is confusing. They aren't accessing with the generated code also. They are interacting with the client we wrote.

Reply:
What if we leave `SetMetadata` and get rid of `EnableMetadata` ? (for example we can add `initialMetadata` field to `Config` instead)

Reply:
Either one is fine to get rid of, if any at all. Just depends on what other think on the breaking change. Worst case is we can deprecate `EnableMetadata` and add more clear documentation